scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/JobEmailNotification.java

```diff
@@ -125,50 +126,83 @@ public boolean doCheckAndSend(boolean withAttachment)
+                    sendEmail(withAttachment, false);
+                }
+                break;
             case JOB_CHANGE_PRIORITY:
             case JOB_IN_ERROR:
             case JOB_PAUSED:
-            case JOB_PENDING_TO_FINISHED:
-            case JOB_PENDING_TO_RUNNING:
             case JOB_RESTARTED_FROM_ERROR:
+            case JOB_PENDING_TO_RUNNING:
             case JOB_RESUMED:
-            case JOB_RUNNING_TO_FINISHED:
             case JOB_SUBMITTED:
-                break;
+                if (jobStatusList.contains(eventType.name().toLowerCase()) ||
+                    jobStatusList.contains(eventType.toString().toLowerCase())) {
+                    sendEmail(withAttachment, false);
+                }
             default:
                 logger.trace("Event not in the list of email notification, doing nothing");
                 return false;
         }
-        if (!PASchedulerProperties.EMAIL_NOTIFICATIONS_ENABLED.getValueAsBoolean()) {
-            logger.debug("Notification emails disabled, doing nothing");
-            return false;
-        }
-        if (!jobStatusList.contains(eventType.toString().toLowerCase()) &&
-            !jobStatusList.contains(eventType.name().toLowerCase())) {
-            return false;
-        }
 
+        return true;
+
+    }
+
+    private void sendEmail(boolean withAttachment, boolean withErrors)
+            throws JobEmailNotificationException, IOException, UnknownJobException, PermissionException {
         try {
             if (withAttachment) {
                 String attachment = getAttachment();
                 if (attachment != null) {
-                    sender.sender(getTo(), getSubject(), getBody(), attachment, getAttachmentName());
+                    sender.sender(getTo(), getSubject(withErrors), getBody(), attachment, getAttachmentName());
                     FileUtils.deleteQuietly(new File(attachment));
                 } else {
-                    sender.sender(getTo(), getSubject(), getBody());
+                    sender.sender(getTo(), getSubject(withErrors), getBody());
                 }
             } else {
-                sender.sender(getTo(), getSubject(), getBody());
+                sender.sender(getTo(), getSubject(withErrors), getBody());
             }
-            return true;
-        } catch (EmailException e) {
+
+        } catch (EmailException e)
+
+        {
             throw new JobEmailNotificationException(String.join(",", getTo()),
                                                     "Error sending email: " + e.getMessage(),
                                                     e);
         }
     }
 
+    private boolean hasFaultyTasks() {
+        List<TaskState> tasks = jobState.getTasks();
+        boolean withErrors = tasks.stream()
+                                  .map(task -> task.getStatus().name().toLowerCase())
+                                  .anyMatch(t -> t.equals("Faulty".toLowerCase()));
```

Comment:
why not using JobInfo contained in notification data?

Besides this, we are interested in getNumberOfFailedTasks, getNumberOfFaultyTasks and getNumberOfInErrorTasks

Reply:
@fviale thanks for notifying this. The new commit should have fixed that.